### DISCUSSION TITLE
看到4.34出来了，观察更新日志有UDP相关的内容更新，测了一下还以为支持NAT开放了，，原来是错觉

### DISCUSSION THREAD

**Discussion by 1265578519** [v2fly/v2ray-core#591] [General]
看到4.34出来了，观察更新日志有UDP相关的内容更新，测了一下还以为支持NAT开放了，，原来是错觉

今天新买了一台俄罗斯服务器测试机，部署上v2
服务端4.34，客户端4.33，NAT类型中等，
![image](https://user-images.githubusercontent.com/6442439/104056467-7d0ce300-522b-11eb-9ab1-b5a2e5c75918.png)
服务端4.34，客户端4.34位相同版本，NAT类型也是中等

然后去腾讯云主力机上进行测试
服务端4.33，客户端4.34，NAT类型严格
![image](https://user-images.githubusercontent.com/6442439/104056541-a0d02900-522b-11eb-93a4-951ce1aa90eb.png)

我把这台服务端升级到4.34确认一下，依旧是NAT类型严格
![image](https://user-images.githubusercontent.com/6442439/104056620-c1987e80-522b-11eb-8909-c6485351eb2c.png)

![image](https://user-images.githubusercontent.com/6442439/104056709-decd4d00-522b-11eb-919b-5f30e304f291.png)

突发奇想，腾讯云是内网IP做网卡，难道是网卡内网IP的缘故导致NAT是严格？

我看看今天开的中等的那一台俄罗斯是什么情况。。果然，ifconfig显示为公网ip，所以中等NAT可能和这个有关
![image](https://user-images.githubusercontent.com/6442439/104056727-e7258800-522b-11eb-86b3-f9440b2b50c5.png)

腾讯云的依旧是严格，看到是内网IP的网卡
![image](https://user-images.githubusercontent.com/6442439/104056766-f6a4d100-522b-11eb-9451-33f6f7dbe62f.png)

所以结论是，

V2还没解决NAT开放问题

**Comment by lp123sun** [comment]
没看懂

**Comment by secXsQuared** [comment]
> 看到4.34出来了，观察更新日志有UDP相关的内容更新，测了一下还以为支持NAT开放了，，原来是错觉
> 
> 今天新买了一台俄罗斯服务器测试机，部署上v2
> 服务端4.34，客户端4.33，NAT类型中等，
> ![image](https://user-images.githubusercontent.com/6442439/104056467-7d0ce300-522b-11eb-9ab1-b5a2e5c75918.png)
> 服务端4.34，客户端4.34位相同版本，NAT类型也是中等
> 
> 然后去腾讯云主力机上进行测试
> 服务端4.33，客户端4.34，NAT类型严格
> ![image](https://user-images.githubusercontent.com/6442439/104056541-a0d02900-522b-11eb-93a4-951ce1aa90eb.png)
> 
> 我把这台服务端升级到4.34确认一下，依旧是NAT类型严格
> ![image](https://user-images.githubusercontent.com/6442439/104056620-c1987e80-522b-11eb-8909-c6485351eb2c.png)
> 
> ![image](https://user-images.githubusercontent.com/6442439/104056709-decd4d00-522b-11eb-919b-5f30e304f291.png)
> 
> 突发奇想，腾讯云是内网IP做网卡，难道是网卡内网IP的缘故导致NAT是严格？
> 
> 我看看今天开的中等的那一台俄罗斯是什么情况。。果然，ifconfig显示为公网ip，所以中等NAT可能和这个有关
> ![image](https://user-images.githubusercontent.com/6442439/104056727-e7258800-522b-11eb-86b3-f9440b2b50c5.png)
> 
> 腾讯云的依旧是严格，看到是内网IP的网卡
> ![image](https://user-images.githubusercontent.com/6442439/104056766-f6a4d100-522b-11eb-9451-33f6f7dbe62f.png)
> 
> 所以结论是，
> 
> V2还没解决NAT开放问题

NAT问题有workaround，家里主网关做个透明代理，然后家里第二网关做成wireguard client，指向主网关。VPS上做个wireguard server。然后wireguard 就可以 wireguard over v2ray。需要NAT A/B的机器网关设成第二网关。之前switch在家 NAT D->NAT B就是这么搞的。

不过打字的时候想是不是用dokodoor不需要这么麻烦。

**Comment by SekiBetu** [comment]
maybe you can try this https://github.com/XTLS/Xray-core/releases/tag/v1.3.0
or just use shadowsocks

i tested the Xray-core 1.3.0, vless protocol, it's FullCone.
![image](https://user-images.githubusercontent.com/38283893/107875716-878e5c80-6efc-11eb-8699-bed5bfd0eabc.png)

**Reply by SekiBetu** [reply]
Netch also supported it.
https://github.com/NetchX/Netch/releases/tag/1.7.4

**Reply by 1265578519** [reply]
> Netch也支持它。
> https://github.com/NetchX/Netch/releases/tag/1.7.4

实测不支持，看他们的更新记录，应该需要卸载原有的v2ray并且替换服务端的Xray核心进行安装
（需服务端 Xray-core v1.3.0+）

**Reply by RPRX** [reply]
@1265578519 

我理解一下，所以您是没按说明替换服务端就“实测不支持”？两端软件都支持才能实现 FullCone，这是常理与常识。
